addons/website_sale/models/sale_order.py

```diff
@@ -359,6 +350,41 @@ class SaleOrderLine(models.Model):
     linked_line_id = fields.Many2one('sale.order.line', string='Linked Order Line', domain="[('order_id', '!=', order_id)]", ondelete='cascade')
     option_line_ids = fields.One2many('sale.order.line', 'linked_line_id', string='Options Linked')
 
+    def unlink(self):
+        to_rename = []
```

Comment:
Use recordset instead of array when possible (benefit from prefetch, prevent duplucate when adding with `|`, ...)

This can actually be simplified to just `to_rename = self.linked_line_id` and let the ORM do the mapping.